mas-foundation/src/main/java/com/ca/mas/core/service/MssoClient.java

```diff
@@ -66,6 +69,11 @@ public long processRequest(MASRequest request, ResultReceiver resultReceiver) {
         Context context = appContext;
         Intent intent = new Intent(MssoIntents.ACTION_PROCESS_REQUEST, null, context, MssoService.class);
         intent.putExtra(MssoIntents.EXTRA_REQUEST_ID, requestId);
+
```

Comment:
I would suggest not to detect logout here, it should transparent, the url is created in MASuser.logout, the policymanager use the request url to determine which policy to run.